components/page-templates/B2B/QuickOrderTemplate/QuickOrderTemplate.spec.tsx

```diff
@@ -0,0 +1,80 @@
+const PromoCodeBadgeMock = () => <div data-testid="promo-code-badge-component" />
+
+jest.mock('@/components/b2b/QuickOrderTable/QuickOrderTable', () => () => QuickOrderTableMock())
+jest.mock('@/components/b2b/B2BProductSearch/B2BProductSearch', () => () => B2BProductSearchMock())
+jest.mock('@/components/cart/CartItemList/CartItemList', () => () => CartItemListMock())
+jest.mock('@/components/common/KeyValueDisplay/KeyValueDisplay', () => () => KeyValueDisplayMock())
+jest.mock('@/components/common/PromoCodeBadge/PromoCodeBadge', () => () => PromoCodeBadgeMock())
+
+describe('[components] QuickOrderTemplate', () => {
+  it('should render QuickOrderTemplate Desktop component', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;(useMediaQuery as jest.Mock).mockReturnValueOnce(true)
+    render(<Common {...Common.args} />)
+
+    const quickOrderText = screen.getByText(/quick-order/i)
+    const initiateQuoteButton = screen.getByRole('button', { name: 'initiate-quote' })
+    const checkoutButton = screen.getByRole('button', { name: 'checkout' })
+    const b2bProductComponent = screen.getByTestId('b2b-product-search-component')
+    const promoCodeComponent = screen.getByTestId('promo-code-badge-component')
+    const orderTotalComponent = screen.getByTestId('key-value-display-component')
+
+    expect(quickOrderText).toBeVisible()
+    expect(screen.getByText(/my-account/i)).toBeVisible()
+    expect(b2bProductComponent).toBeVisible()
+
+    await waitFor(() => {
+      const quickOrderTableComponent = screen.queryByTestId('quick-order-table-component')
+      expect(quickOrderTableComponent).toBeVisible()
+    })
+
+    expect(initiateQuoteButton).toBeVisible()
+    expect(checkoutButton).toBeVisible()
+    expect(promoCodeComponent).toBeVisible()
+    expect(orderTotalComponent).toBeVisible()
+  })
+
+  it('should render QuickOrderTemplate Mobile component', async () => {
+    // eslint-disable-next-line @typescript-eslint/no-extra-semi
+    ;(useMediaQuery as jest.Mock).mockReturnValueOnce(false)
+
+    render(<QuickOrderTemplateMobile {...QuickOrderTemplateMobile.args} />)
+    const quickOrderText = screen.getByText(/quick-order/i)
+    const b2bProductComponent = screen.getByTestId('b2b-product-search-component')
+
+    expect(screen.getByText(/my-account/i)).toBeVisible()
+    expect(b2bProductComponent).toBeVisible()
+    expect(quickOrderText).toBeVisible()
+
+    await waitFor(() => {
+      const cartItemListComponents = screen.queryByTestId('cart-item-list-component')
+      expect(cartItemListComponents).toBeVisible()
+    })
+    await waitFor(() => {
+      const initiateQuoteButton = screen.queryByRole('button', { name: /initiate-quote/i })
+      expect(initiateQuoteButton).toBeVisible()
+    })
+    await waitFor(() => {
+      const checkoutButton = screen.queryByRole('button', { name: /checkout/i })
+      expect(checkoutButton).toBeVisible()
+    })
+  })
+})
```

Comment:
**Missing Testcases**
should add product in list when user clicks on non configurable product 
should display configure dialogue when user clicks on configurable product
should apply coupon when user enters valid coupon code and clicks on Apply button
should remove coupon when user removes removes it

Reply:
Resolved